fix:empty values must be accepted in extrafields

bafbes opened the pull request:
# Fix #[_empty values must be accepted in extrafields_]

bafbes committed:
fix:empty values must be accepted in extrafields

eldy commented on htdocs/core/class/commonobject.class.php:
```diff
@@ -4250,7 +4250,7 @@ function updateExtraField($key)
 		if (! empty($conf->global->MAIN_EXTRAFIELDS_DISABLED)) return 0;	// For avoid conflicts if trigger used
 
-        if (! empty($this->array_options) && !empty($this->array_options["options_$key"]))
+        if (! empty($this->array_options) )
```
I think correct fix, should be 
if (! empty($this->array_options) && isset($this->array_options["options_$key"]))

Does this other suggestion work for your case ?

bafbes committed:
enhance test

In thi case, I think there is no need to test also the non emptiness of the entire table

bafbes commented:
In this case, I think there is no need to test also the non emptiness of the entire table.

bafbes commented:
Thank you for this correction.
It has been a real confusion from my part as I was not allowing emptiness in my test.
Whatever I don't see to what serves testing the non emptiness of the entire array.
Isn't it sufficient to test with :  if ( isset($this->array_options["options_".$key]))
I tried it and found that it does not generate any error in the case array_options does not exist.
